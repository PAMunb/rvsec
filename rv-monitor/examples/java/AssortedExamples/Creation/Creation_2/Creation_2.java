package Creation_2;

public class Creation_2 {
	public int temp = 0;
	public void fun1(){
		temp = 1;
	}
	public void fun2(){
		temp = 2;
	}

	public static void main(String[] args){
		Creation_2 o = new Creation_2();

		System.out.println("fun2");
		o.fun2();
		rvm.CreationRuntimeMonitor.fun2Event(o);
		System.out.println("fun2");
		o.fun2();
		rvm.CreationRuntimeMonitor.fun2Event(o);
		System.out.println("fun2");
		o.fun2();
		rvm.CreationRuntimeMonitor.fun2Event(o);

		System.out.println("main end");
		rvm.CreationRuntimeMonitor.mainendEvent();
	}
}



