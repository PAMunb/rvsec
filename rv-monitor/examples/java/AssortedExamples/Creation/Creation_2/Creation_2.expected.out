fun2
fun2
fun2
main end
